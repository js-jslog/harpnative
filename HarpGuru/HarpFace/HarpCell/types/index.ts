import type { TextStyle, ViewStyle } from 'react-native'

export type Coord = number

export type YXCoord = [Coord, Coord]

export type HarpCellProps = {
  readonly yxCoord: YXCoord
}

export type HarpCellStyles = {
  readonly cell: ViewStyle
  readonly note: TextStyle
  readonly modifier: TextStyle
}
