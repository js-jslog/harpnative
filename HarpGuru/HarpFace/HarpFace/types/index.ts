import { ViewStyle } from 'react-native'

export type HarpFaceStyles = {
  readonly facewrapper: ViewStyle;
  readonly face: ViewStyle;
}
