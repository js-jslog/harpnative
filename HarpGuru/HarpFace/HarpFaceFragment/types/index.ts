import type { ViewStyle } from 'react-native'

import type { HarpRowProps } from '../../HarpRow'

export type HarpFaceFragmentProps = Omit<HarpRowProps, 'yCoord'>

export type FragmentFacts = {
  readonly columnCount: number
  readonly rowCount: number
}

export type HarpFaceFragmentStyles = {
  readonly fragment: ViewStyle
}
