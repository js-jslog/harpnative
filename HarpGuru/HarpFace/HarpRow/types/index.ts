import type { ViewStyle } from 'react-native'

import type { HarpCellProps } from '../../HarpCell'

export type Coord = number

export type HarpRowProps = Pick<HarpCellProps, 'activeHarpStrata' | 'setActiveHarpStrata' | 'activeDisplayMode' | 'activeDisplayMode'> & {
  readonly yCoord: Coord
  readonly xRange: ReadonlyArray<number>
}

export type HarpRowStyles = {
  readonly row: ViewStyle
}
