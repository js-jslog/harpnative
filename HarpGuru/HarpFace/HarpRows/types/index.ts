import type { ReactElement } from 'react'

export type HarpRows = {
  readonly top: ReactElement[]
  readonly bottom: ReactElement[]
}
