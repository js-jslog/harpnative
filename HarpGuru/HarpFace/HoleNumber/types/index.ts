import type { ViewStyle, TextStyle } from 'react-native'

import type { Coord } from '../../HarpRow'

export enum HoleNumberIds {
  One = '1'      , Two = '2'      , Three = '3'   , Four = '4',
  Five = '5'     , Six = '6'      , Seven = '7'   , Eight = '8',
  Nine = '9'     , Ten = '10'     , Eleven = '11' , Twelve = '12',
  Thirteen = '13', Fourteen = '14', Fifteen = '15', Sixteen = '16',
}

export type HoleNumberProps = {
  readonly xCoord: Coord;
}

export type HoleNumberStyles = {
  readonly cell: ViewStyle;
  readonly text: TextStyle;
}
