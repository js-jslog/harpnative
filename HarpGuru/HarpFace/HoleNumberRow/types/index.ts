import type { ViewStyle } from 'react-native'

import type { HarpFaceFragmentProps } from '../../HarpFaceFragment'

export type HoleNumberRowProps = Pick<HarpFaceFragmentProps, 'xRange'>

export type HoleNumberRowStyles = {
  readonly row: ViewStyle
}
