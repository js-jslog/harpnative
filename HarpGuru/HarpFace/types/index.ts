import type { HarpFaceFragmentProps } from '../HarpFaceFragment'

export enum DisplayModes {
  Degree = 'DEGREE',
  Pitch = 'PITCH',
}

export type HarpFaceProps = Omit<HarpFaceFragmentProps, 'xRange'>
