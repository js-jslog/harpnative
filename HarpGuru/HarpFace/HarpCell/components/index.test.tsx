import reactn from 'reactn'
import React from 'react'
import { PitchIds, DegreeIds } from 'harpstrata'
import { render, fireEvent } from '@testing-library/react-native'

import {
  inactiveCellsHarpFaceProps,
  activeCellsHarpFaceProps,
} from '../../testResources'
import { DisplayModes } from '../../../types'

import { HarpCell } from './index'

jest.mock('reactn')

test('A component is rendered with the Degree or Pitch value in its text view depending on the DisplayMode selected', () => {
  reactn.useGlobal.mockReturnValue([inactiveCellsHarpFaceProps.activeHarpStrata])
  const harpCellProps = {
    activeDisplayMode: DisplayModes.Degree,
    yxCoord: [3, 0] as [3, 0],
  }
  const { getByText, rerender } = render(<HarpCell {...harpCellProps} />)

  expect(getByText(DegreeIds.Second)).toBeTruthy()

  rerender(
    <HarpCell {...harpCellProps} activeDisplayMode={DisplayModes.Pitch} />
  )

  expect(getByText(PitchIds.D)).toBeTruthy()
})

test('A component is rendered with an a11y role of button', () => {
  reactn.useGlobal.mockReturnValue([inactiveCellsHarpFaceProps.activeHarpStrata])
  const harpCellProps = {
    activeDisplayMode: DisplayModes.Degree,
    yxCoord: [3, 0] as [3, 0],
  }
  const { getByRole } = render(<HarpCell {...harpCellProps} />)

  expect(getByRole('button')).toBeTruthy()
})

test('A component is rendered without an a11y role of button if it has no content', () => {
  reactn.useGlobal.mockReturnValue([inactiveCellsHarpFaceProps.activeHarpStrata])
  const harpCellProps = {
    activeDisplayMode: DisplayModes.Degree,
    yxCoord: [0, 0] as [0, 0],
  }
  const { queryByRole } = render(<HarpCell {...harpCellProps} />)

  expect(queryByRole('button')).toBeNull()
})

test.skip('A press of the componenet results in toggled active ids in the harpstrata passed to the paramaterised setter', () => {
  const setActiveHarpStrata = jest.fn()
  const harpCellProps = {
    ...inactiveCellsHarpFaceProps,
    activeDisplayMode: DisplayModes.Degree,
    setActiveHarpStrata,
    yxCoord: [3, 0] as [3, 0],
  }

  const { getByText } = render(<HarpCell {...harpCellProps} />)

  fireEvent.press(getByText(DegreeIds.Second))

  const {
    mock: {
      calls: [[newHarpStrata]],
    },
  } = setActiveHarpStrata
  const {
    isActiveComplex: {
      activeDegreeIds: newDegreeIds,
      activePitchIds: newPitchIds,
    },
  } = newHarpStrata

  const expectedDegreeIds = [DegreeIds.Second]
  const expectedPitchIds = [PitchIds.D]

  expect(setActiveHarpStrata.mock.calls.length).toBe(1)
  expect(newDegreeIds).toStrictEqual(expectedDegreeIds)
  expect(newPitchIds).toStrictEqual(expectedPitchIds)
})

test('A snapshot of a populated cell', () => {
  reactn.useGlobal.mockReturnValue([inactiveCellsHarpFaceProps.activeHarpStrata])
  const harpCellProps = {
    activeDisplayMode: DisplayModes.Degree,
    yxCoord: [3, 0] as [3, 0],
  }

  const { container } = render(<HarpCell {...harpCellProps} />)
  expect(container).toMatchSnapshot()
})

test('A snapshot of an active cell', () => {
  reactn.useGlobal.mockReturnValue([activeCellsHarpFaceProps.activeHarpStrata])
  const harpCellProps = {
    activeDisplayMode: DisplayModes.Degree,
    yxCoord: [3, 0] as [3, 0],
  }

  const { container } = render(<HarpCell {...harpCellProps} />)
  expect(container).toMatchSnapshot()
})

test('A snapshot of an empty cell', () => {
  reactn.useGlobal.mockReturnValue([inactiveCellsHarpFaceProps.activeHarpStrata])
  const harpFaceProps = {
    activeDisplayMode: DisplayModes.Degree,
    yxCoord: [0, 0] as [0, 0],
  }

  const { container } = render(<HarpCell {...harpFaceProps} />)
  expect(container).toMatchSnapshot()
})
