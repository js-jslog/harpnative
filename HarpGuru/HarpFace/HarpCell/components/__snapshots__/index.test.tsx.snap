// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A snapshot of a populated cell 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fefefe",
        "borderColor": "#ced4da",
        "borderRadius": 7.5394000000000005,
        "borderWidth": 1.1,
        "elevation": 0,
        "height": 31.9374,
        "justifyContent": "center",
        "width": 31.9374,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ced4da",
          "fontSize": 12.199000000000002,
        }
      }
    >
      2
    </Text>
  </View>
</View>
`;

exports[`A snapshot of an active cell 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#323031",
        "borderColor": "#323031",
        "borderRadius": 7.5394000000000005,
        "borderWidth": 1.1,
        "elevation": 1.7798000000000003,
        "height": 31.9374,
        "justifyContent": "center",
        "width": 31.9374,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fefefe",
          "fontSize": 12.199000000000002,
        }
      }
    >
      2
    </Text>
  </View>
</View>
`;

exports[`A snapshot of an empty cell 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fefefe",
        "borderColor": "#ced4da",
        "borderRadius": 7.5394000000000005,
        "borderWidth": 0,
        "elevation": 0,
        "height": 31.9374,
        "justifyContent": "center",
        "width": 31.9374,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fefefe",
          "borderColor": "#ced4da",
          "borderRadius": 7.5394000000000005,
          "borderWidth": 0,
          "elevation": 0,
          "height": 31.9374,
          "justifyContent": "center",
          "width": 31.9374,
        }
      }
    />
  </View>
</View>
`;
